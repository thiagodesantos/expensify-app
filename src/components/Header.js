import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({startLogout}) => (
    <header>
        <h1>Despesinhas</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
        <NavLink to="/criar" activeClassName="is-active">Adicionar</NavLink>
        <NavLink to="/ajuda" activeClassName="is-active">Ajuda</NavLink>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);