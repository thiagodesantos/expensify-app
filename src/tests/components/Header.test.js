import React from 'react';
import { shallow } from 'enzyme';
//import toJson from 'enzyme-to-json';
import Header from '../../components/Header';

test('Deveria renderizar o header corretamente', () => {
    const wrapper = shallow( <Header />);
    expect(wrapper).toMatchSnapshot();

    //expect(wrapper.find('h1').text()).toBe('Despesinhas');
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});
