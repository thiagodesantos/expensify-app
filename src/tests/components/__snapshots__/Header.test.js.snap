// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deveria renderizar o header corretamente 1`] = `
<header>
  <h1>
    Despesinhas
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/criar"
  >
    Adicionar
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/ajuda"
  >
    Ajuda
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
